string state
string grasped_item
